batchDivision/batch_division.go

```diff
@@ -0,0 +1,80 @@
+package batchDivision
+
+import (
+	"github.com/codeuniversity/al-proto"
+	"math"
+	"math/rand"
+	"strconv"
+	"time"
+)
+
+func createBatches(cells []*proto.Cell, batchSize uint) map[string][]*proto.Cell {
+	dict := make(map[string][]*proto.Cell)
```

Comment:
I would add a type for that map
i.e. something like
```golang
type Buckets map[string][]*proto.Cell
```
(I would use `Buckets` instead of `Batches` to make sure we don't get confused with the batches we send to cis)